### ISSUE TITLE
when we use rocket.chat 3.9.4 version,without (USE_NATIVE_OPLOG=true),our mongodb will have so many getmore operations and the mongodb's cpu is 99%......

### ISSUE THREAD

**Issue by codeneno** [RocketChat/Rocket.Chat#20027]
### Steps to reproduce:

1. 
2. 
3. 

### Server Setup Information:

- Version of Rocket.Chat Server: 3.9.4
- Operating System: ubuntu
- Deployment Method: docker
- Number of Running Instances: 22
- DB Replicaset Oplog: enabled
- NodeJS Version: 12.*
- MongoDB Version:4.2.1

### Client Setup Information

- Desktop App or Browser Version: chrome 81
- Operating System: windows

### Additional context

when we use rocket.chat 3.9.4 version,without (USE_NATIVE_OPLOG=true),our mongodb will have much getmore operation,
and the mongodb's cpu is 99%......

![image](https://user-images.githubusercontent.com/66900278/103476261-1b91e200-4def-11eb-963f-bdfd75c11721.png)

![image](https://user-images.githubusercontent.com/66900278/103476314-7deae280-4def-11eb-868e-5bcd96950357.png)

just for example 

```
{
    "op": "getmore",
    "ns": "rocketchat.rocketchat_permissions",
    "command": {
        "getMore": {
            "$numberLong": "708537366840158110"
        },
        "collection": "rocketchat_permissions",
        "batchSize": 1000,
        "lsid": {
            "id": {
                "$binary": "VwkQAET6Q76OrpFkoCIBNg==",
                "$type": "04"
            }
        },
        "$db": "rocketchat"
    },
    "originatingCommand": {
        "aggregate": "rocketchat_permissions",
        "pipeline": [
            {
                "$changeStream": {
                    "fullDocument": "default"
                }
            }
        ],
        "cursor": {},
        "lsid": {
            "id": {
                "$binary": "VwkQAET6Q76OrpFkoCIBNg==",
                "$type": "04"
            }
        },
        "$db": "rocketchat"
    },
    "cursorid": {
        "$numberLong": "708537366840158110"
    },
    "keysExamined": 0,
    "docsExamined": 12,
    "numYield": 44,
    "nreturned": 0,
    "locks": {
        "ParallelBatchWriterMode": {
            "acquireCount": {
                "r": {
                    "$numberLong": "1"
                }
            }
        },
        "ReplicationStateTransition": {
            "acquireCount": {
                "w": {
                    "$numberLong": "47"
                }
            }
        },
        "Global": {
            "acquireCount": {
                "r": {
                    "$numberLong": "47"
                }
            }
        },
        "Database": {
            "acquireCount": {
                "r": {
                    "$numberLong": "46"
                }
            }
        },
        "Mutex": {
            "acquireCount": {
                "r": {
                    "$numberLong": "2"
                }
            }
        },
        "oplog": {
            "acquireCount": {
                "r": {
                    "$numberLong": "45"
                }
            }
        }
    },
    "flowControl": {},
    "storage": {},
    "responseLength": 190,
    "protocol": "op_msg",
    "millis": 103,
    "planSummary": "COLLSCAN"
}
```

**Comment by codeneno** [comment]
BUT WHEN WE USE_NATIVE_OPTLOG=true, there are so may logs:

2021-01-03T10:31:29.315826883Z   userId: '4K9d7sde32ieAzRBZ'
2021-01-03T10:31:29.315829658Z }
2021-01-03T10:31:32.480555250Z null {
2021-01-03T10:31:32.480584876Z   id: 'YF3a68esrJMdJHQES',
2021-01-03T10:31:32.480588714Z   clientAddress: null,
2021-01-03T10:31:32.480591740Z   httpHeaders: {
2021-01-03T10:31:32.480594828Z     'x-real-ip': '10.129.21.153',
2021-01-03T10:31:32.480597952Z     host: 'nwt.icbf.com.cn',
2021-01-03T10:31:32.480601120Z     'user-agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/71.0.3578.98 Safari/537.36',
2021-01-03T10:31:32.480604213Z     'accept-language': 'zh-CN,zh;q=0.9'
2021-01-03T10:31:32.480606991Z   },
2021-01-03T10:31:32.480609711Z   userId: '4K9d7sde32ieAzRBZ'
2021-01-03T10:31:32.480612555Z }
2021-01-03T10:32:18.395154914Z null {
2021-01-03T10:32:18.395187968Z   id: 'YuWsYhrEw7Tu3EGgQ',
2021-01-03T10:32:18.395192331Z   clientAddress: null,
2021-01-03T10:32:18.395195894Z   httpHeaders: {
2021-01-03T10:32:18.395199194Z     'x-real-ip': '10.129.21.141',
2021-01-03T10:32:18.395202499Z     host: 'nwt.icbf.com.cn',
2021-01-03T10:32:18.395206027Z     'user-agent': 'Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.88 Safari/537.36',
2021-01-03T10:32:18.395209454Z     'accept-language': 'zh-CN,zh;q=0.9'
2021-01-03T10:32:18.395212606Z   },
2021-01-03T10:32:18.395215647Z   userId: null
2021-01-03T10:32:18.395218845Z }
2021-01-03T10:32:18.716742529Z null {
2021-01-03T10:32:18.716770899Z   id: 'YuWsYhrEw7Tu3EGgQ',
2021-01-03T10:32:18.716775127Z   clientAddress: null,
2021-01-03T10:32:18.716778498Z   httpHeaders: {
2021-01-03T10:32:18.716781912Z     'x-real-ip': '10.129.21.141',
2021-01-03T10:32:18.716785272Z     host: 'nwt.icbf.com.cn',
2021-01-03T10:32:18.716788614Z     'user-agent': 'Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.88 Safari/537.36',
2021-01-03T10:32:18.716806674Z     'accept-language': 'zh-CN,zh;q=0.9'
2021-01-03T10:32:18.716809886Z   },
2021-01-03T10:32:18.716812918Z   userId: 'o9X7Wcf6JFa62pAwy'
2021-01-03T10:32:18.716815802Z }
2021-01-03T10:42:59.487757615Z null {
2021-01-03T10:42:59.487792790Z   id: 'WYpHASXdoJEpjxENH',
2021-01-03T10:42:59.487796952Z   clientAddress: null,
2021-01-03T10:42:59.487800182Z   httpHeaders: {
2021-01-03T10:42:59.487803255Z     'x-real-ip': '10.129.21.146',
2021-01-03T10:42:59.487806376Z     host: 'nwt.icbf.com.cn',
2021-01-03T10:42:59.487809403Z     'user-agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36',
2021-01-03T10:42:59.487812528Z     'accept-language': 'zh-CN,zh;q=0.9'
2021-01-03T10:42:59.487815432Z   },
2021-01-03T10:42:59.487818274Z   userId: null
2021-01-03T10:42:59.487821112Z }
2021-01-03T10:42:59.501392219Z null {
2021-01-03T10:42:59.501416374Z   id: 'WYpHASXdoJEpjxENH',
2021-01-03T10:42:59.501420722Z   clientAddress: null,
2021-01-03T10:42:59.501423855Z   httpHeaders: {
2021-01-03T10:42:59.501427217Z     'x-real-ip': '10.129.21.146',
2021-01-03T10:42:59.501430164Z     host: 'nwt.icbf.com.cn',
2021-01-03T10:42:59.501433299Z     'user-agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36',
2021-01-03T10:42:59.501436448Z     'accept-language': 'zh-CN,zh;q=0.9'
2021-01-03T10:42:59.501439423Z   },
2021-01-03T10:42:59.501442233Z   userId: 'BzeGsCECSmFQXBEAg'
2021-01-03T10:42:59.501445102Z }

**Comment by emikolajczak** [comment]
I can confirm flood of getmore operations on MongoDB when "USE_NATIVE_OPLOG=true" is not set.

**Comment by 564064202** [comment]
may be it's a bug

**Comment by 564064202** [comment]
> I can confirm flood of getmore operations on MongoDB when "USE_NATIVE_OPLOG=true" is not set.
 we did

**Comment by ankar84** [comment]
> BUT WHEN WE USE_NATIVE_OPTLOG=true, there are so may logs:
> 
> ```
> 2021-01-03T10:31:29.315826883Z   userId: '4K9d7sde32ieAzRBZ'
> 2021-01-03T10:31:29.315829658Z }
> 2021-01-03T10:31:32.480555250Z null {
> 2021-01-03T10:31:32.480584876Z   id: 'YF3a68esrJMdJHQES',
> 2021-01-03T10:31:32.480588714Z   clientAddress: null,
> 2021-01-03T10:31:32.480591740Z   httpHeaders: {
> 2021-01-03T10:31:32.480594828Z     'x-real-ip': '10.129.21.153',
> 2021-01-03T10:31:32.480597952Z     host: 'nwt.icbf.com.cn',
> 2021-01-03T10:31:32.480601120Z     'user-agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/71.0.3578.98 Safari/537.36',
> 2021-01-03T10:31:32.480604213Z     'accept-language': 'zh-CN,zh;q=0.9'
> 2021-01-03T10:31:32.480606991Z   },
> 2021-01-03T10:31:32.480609711Z   userId: '4K9d7sde32ieAzRBZ'
> 2021-01-03T10:31:32.480612555Z }
> 2021-01-03T10:32:18.395154914Z null {
> 2021-01-03T10:32:18.395187968Z   id: 'YuWsYhrEw7Tu3EGgQ',
> 2021-01-03T10:32:18.395192331Z   clientAddress: null,
> 2021-01-03T10:32:18.395195894Z   httpHeaders: {
> 2021-01-03T10:32:18.395199194Z     'x-real-ip': '10.129.21.141',
> 2021-01-03T10:32:18.395202499Z     host: 'nwt.icbf.com.cn',
> 2021-01-03T10:32:18.395206027Z     'user-agent': 'Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.88 Safari/537.36',
> 2021-01-03T10:32:18.395209454Z     'accept-language': 'zh-CN,zh;q=0.9'
> 2021-01-03T10:32:18.395212606Z   },
> 2021-01-03T10:32:18.395215647Z   userId: null
> 2021-01-03T10:32:18.395218845Z }
> 2021-01-03T10:32:18.716742529Z null {
> 2021-01-03T10:32:18.716770899Z   id: 'YuWsYhrEw7Tu3EGgQ',
> 2021-01-03T10:32:18.716775127Z   clientAddress: null,
> 2021-01-03T10:32:18.716778498Z   httpHeaders: {
> 2021-01-03T10:32:18.716781912Z     'x-real-ip': '10.129.21.141',
> 2021-01-03T10:32:18.716785272Z     host: 'nwt.icbf.com.cn',
> 2021-01-03T10:32:18.716788614Z     'user-agent': 'Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.88 Safari/537.36',
> 2021-01-03T10:32:18.716806674Z     'accept-language': 'zh-CN,zh;q=0.9'
> 2021-01-03T10:32:18.716809886Z   },
> 2021-01-03T10:32:18.716812918Z   userId: 'o9X7Wcf6JFa62pAwy'
> 2021-01-03T10:32:18.716815802Z }
> 2021-01-03T10:42:59.487757615Z null {
> 2021-01-03T10:42:59.487792790Z   id: 'WYpHASXdoJEpjxENH',
> 2021-01-03T10:42:59.487796952Z   clientAddress: null,
> 2021-01-03T10:42:59.487800182Z   httpHeaders: {
> 2021-01-03T10:42:59.487803255Z     'x-real-ip': '10.129.21.146',
> 2021-01-03T10:42:59.487806376Z     host: 'nwt.icbf.com.cn',
> 2021-01-03T10:42:59.487809403Z     'user-agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36',
> 2021-01-03T10:42:59.487812528Z     'accept-language': 'zh-CN,zh;q=0.9'
> 2021-01-03T10:42:59.487815432Z   },
> 2021-01-03T10:42:59.487818274Z   userId: null
> 2021-01-03T10:42:59.487821112Z }
> 2021-01-03T10:42:59.501392219Z null {
> 2021-01-03T10:42:59.501416374Z   id: 'WYpHASXdoJEpjxENH',
> 2021-01-03T10:42:59.501420722Z   clientAddress: null,
> 2021-01-03T10:42:59.501423855Z   httpHeaders: {
> 2021-01-03T10:42:59.501427217Z     'x-real-ip': '10.129.21.146',
> 2021-01-03T10:42:59.501430164Z     host: 'nwt.icbf.com.cn',
> 2021-01-03T10:42:59.501433299Z     'user-agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36',
> 2021-01-03T10:42:59.501436448Z     'accept-language': 'zh-CN,zh;q=0.9'
> 2021-01-03T10:42:59.501439423Z   },
> 2021-01-03T10:42:59.501442233Z   userId: 'BzeGsCECSmFQXBEAg'
> 2021-01-03T10:42:59.501445102Z }
> ```

Yes! We have a lot of these since moved to native oplog.

**Comment by sampaiodiego** [comment]
this has been fixed by https://github.com/RocketChat/Rocket.Chat/pull/26336 and released on version `5.0.1`.. 

actually `USE_NATIVE_OPTLOG` is now broken on 5.x .. waiting on this issue to be closed: https://github.com/meteor/meteor/issues/11993 it has been already fixed on latest mongodb nodejs driver, just needs a new release from Meteor.
